src/main/java/soconnect/model/person/search/ContactMightBeRelevantPredicate.java

```diff
@@ -1,43 +1,70 @@
-        this.keywords = keywords;
+    public ContactMightBeRelevantPredicate(ArgumentMultimap argMultimap) {
+        this.argMultimap = argMultimap;
     }
 
     @Override
     public boolean test(Person person) {
-        boolean isNameContained = keywords.stream()
-                .allMatch(keyword -> StringUtil.containsSomeKeywordsIgnoreCase(
-                        person.getName().fullName, keyword
-                ));
-        boolean isAddressContained = keywords.stream()
-                .allMatch(keyword -> StringUtil.containsSomeKeywordsIgnoreCase(
-                        person.getAddress().value, keyword));
-        boolean isEmailContained = keywords.stream()
-                .allMatch(keyword -> StringUtil.containsSomeKeywordsIgnoreCase(
-                        person.getEmail().value, keyword));
-        boolean isPhoneContained = keywords.stream()
-                .allMatch(keyword -> StringUtil.containsSomeKeywordsIgnoreCase(
-                        person.getPhone().value, keyword));
-        boolean isTagContained = keywords.stream()
-                .allMatch(keyword -> person.getTags().stream()
-                        .anyMatch(tag -> StringUtil.containsSomeKeywordsIgnoreCase(tag.tagName, keyword)));
+        for (Prefix prefix : argMultimap.getAllPrefixes()) {
+            SearchPrefix.SearchPrefixCommand prefixCommand = convertPrefixToEnumType(prefix);
+            List<String> keywords = argMultimap.getAllValues(prefix);
+            switch (prefixCommand) {
+            case NAME:
+                isNameContained = keywords.stream()
+                        .anyMatch(keyword -> StringUtil
+                                .containsSomeKeywordsIgnoreCase(person.getName().fullName, keyword));
+                break;
+            case ADDRESS:
+                isAddressContained = keywords.stream()
+                        .anyMatch(keyword -> StringUtil
+                                .containsSomeKeywordsIgnoreCase(person.getAddress().value, keyword));
+                break;
+            case EMAIL:
+                isEmailContained = keywords.stream()
+                        .anyMatch(keyword -> StringUtil
+                                .containsSomeKeywordsIgnoreCase(person.getEmail().value, keyword));
+                break;
+            case PHONE:
+                isPhoneContained = keywords.stream()
+                        .anyMatch(keyword -> StringUtil
+                                .containsSomeKeywordsIgnoreCase(person.getPhone().value, keyword));
+                break;
+            case TAG:
+                isTagContained = keywords.stream()
+                        .anyMatch(keyword -> person.getTags().stream()
+                                .anyMatch(tag -> StringUtil
+                                        .containsSomeKeywordsIgnoreCase(tag.tagName, keyword)));
+                break;
+            default:
+                break;
```

Comment:
I think maybe u could throw an exception. Or use assertions instead of leaving it as just `break`.

Reply:
Actually, if I am not wrong, I cannot really throw exception because it is already outside of the parser class. And it is not suitable because this class is inheriting the Predicate class and the method `test` is overriding the parent `test` method. I tried but couldn't. Therefore I designed it in such a way that it will never actually reach the default case. This is because the parser only parse the argument like this:
`        ArgumentMultimap argMultimap =
                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);`

This will cause the tokenizer to only tokenize the valid prefixes. 

I tried using assertions also, but it cause the valid search to also cause error :")

Please suggest if you have idea to double defense it safely :)

Reply:
Hey, update! Although I am still unable to throw exception, thanks to this I discover some bugs and manage to turn it into improvement in relevant search instead :DD

So since the `argMultimap` will just tokenize the valid prefixes, when we feed in args like `search n/alex h/yeoh`, even though the `h/yeoh` is invalid, it will map it as `[prefix_name, alex h/yeoh]`, aka it will treat the entirety as name. This instead is not handled well by the pervious search algorithm, as it will just return empty list of person.

However, now it will return `alex yeoh` instead! And even if the user types in `search n/a+l+e+x`, or some other weird combinations, it will still be able to return relevant result. Contrastingly, if the user types in `search f/hello` or `search n/++`, then it will return the `invalid_format` error.

So thank you! :D

Reply:
Lol. That's great! Keep up the good work!